services/user.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth import get_user_model
+
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> User:
+    user = get_user_model().objects.create_user(username=username)
```

Comment:
The `create_user` function already handles password hashing, you don't need to call `set_password` method separately. You can pass the password directly in `create_user` method.